test/e2e/util_test.go

```diff
@@ -918,3 +919,30 @@ func HaveMessage(goal string) gtypes.GomegaMatcher {
 		return plan.Status.Message
 	}, ContainSubstring(goal))
 }
+
+func SetUpGeneratedTestNamespace(name string) (string, corev1.Namespace) {
+	generatedNamespace := genName(name)
+	ns := corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
Mainly curious on when this is necessary: do we need to specify TypeMeta here as well?